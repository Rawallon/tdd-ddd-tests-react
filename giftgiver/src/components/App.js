import React, { useState } from 'react';
import { Button } from 'react-bootstrap';
import Gift from './Gift';

export default function App() {
  const [state, setstate] = useState([]);
  const addGift = () => {
    setstate((prev) => [...prev, { id: state.length + 1 }]);
  };
  const removeGift = (propId) => {
    setstate(state.filter((gift) => gift.id !== propId));
  };
  return (
    <div>
      <h2>
        Gift Giver <span>({state.length})</span>
      </h2>
      <div className="gift-list">
        {state.map((item) => (
          <Gift key={item.id}>
            <Button className="btn-rem" onClick={() => removeGift(item.id)}>
              Remove Gift
            </Button>
          </Gift>
        ))}
      </div>
      <Button className="btn-add" onClick={addGift}>
        Add gift
      </Button>
    </div>
  );
}
