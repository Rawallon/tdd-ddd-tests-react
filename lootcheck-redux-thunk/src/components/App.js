import React from 'react';
import Loot from './Loot';
import Wallet from './Wallet';

export default function App() {
  return (
    <div>
      <h2>Loot Check</h2>
      <Wallet />
      <Loot />
    </div>
  );
}
