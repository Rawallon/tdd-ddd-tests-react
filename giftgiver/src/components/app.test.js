import React from 'React';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const app = shallow(<App />);

  it('Render correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('Start with no gifts', () => {
    expect(app.find('.gift-list').children().length).toEqual(0);
  });

  describe('When clicking', () => {
    it('Add new gift when clicking the button', () => {
      app.find('.btn-add').simulate('click');
      expect(app.find('span').at(0).text()).toEqual('(1)');
      expect(app.find('.gift-list').children().length).toEqual(1);
    });

    it('Renders Gift component', () => {
      expect(app.find('Gift').exists()).toBe(true);
    });

    it('Removes a Gift component', () => {
      app.find('.btn-rem').simulate('click');
      expect(app.find('.gift-list').children().length).toEqual(0);
    });
  });
});
