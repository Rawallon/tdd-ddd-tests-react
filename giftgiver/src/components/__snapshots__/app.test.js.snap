// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render correctly 1`] = `ShallowWrapper {}`;
